sieve.js

```diff
@@ -472,99 +470,115 @@
-                "Type": "If"
+                Then: parameters.actions,
+                Type: 'If'
             }
         ];
+
+        if (parameters.vacation) {
+            treeStructure.push(parameters.vacation);
+        }
+
+        return treeStructure;
     }
 
     function buildTestNegate(test) {
         return {
-            "Test": test,
-            "Type": "Not"
+            Test: test,
+            Type: 'Not'
         };
     }
 
-    function buildSieveRequire()
+    function buildSieveRequire(requires)
     {
         return {
-            "List": ["fileinto", "imap4flags"],
-            "Type": "Require"
+            List: ['fileinto', 'imap4flags'].concat(requires),
+            Type: 'Require'
         };
     }
 
     function buildHeaderTest(headers, keys, match) {
         return {
-            "Headers": headers,
-            "Keys": keys,
-            "Match":
-            {
-                "Type": match
+            Headers: headers,
+            Keys: keys,
+            Match: {
+                Type: match
             },
-            "Format":
-            {
-                "Type": "UnicodeCaseMap"
+            Format: {
+                Type: 'UnicodeCaseMap'
             },
-            "Type": "Header"
+            Type: 'Header'
         };
     }
 
     function buildAddressTest(headers, keys, match) {
-        addresspart = "All"; // FIXME Matching to the whole address for now
+        addresspart = 'All'; // FIXME Matching to the whole address for now
         return {
-            "Headers": headers,
-            "Keys": keys,
-            "Match":
-            {
-                "Type": match
+            Headers: headers,
+            Keys: keys,
+            Match: {
+                Type: match
             },
-            "Format":
-            {
-                "Type": "UnicodeCaseMap"
+            Format: {
+                Type: 'UnicodeCaseMap'
             },
-            "Type": "Address",
-            "AddressPart":
-            {
-                "Type": addresspart
+            Type: 'Address',
+            AddressPart: {
+                Type: addresspart
             }
         };
     }
 
     function buildAttachmentTest() {
         return {
-            "Headers":["X-Attached"],
-            "Type":"Exists"
+            Headers:['X-Attached'],
+            Type:'Exists'
         };
     }
 
     function buildSetflagThen(read, starred) {
         flags = [];
         if (read) {
-            flags.push("\\Seen");
+            flags.push('\\Seen');
         }
         if (starred) {
-            flags.push("\\Flagged");
+            flags.push('\\Flagged');
         }
         return {
-            "Flags": flags,
-            "Type": "AddFlag"
+            Flags: flags,
+            Type: 'AddFlag'
+        };
+    }
+
+    function buildVacation(message) {
+        return {
+            Message: message,
+            Args: {
+                subject: DEFAULT_VACATION_SUBJECT
```

Comment:
If we just want to use the default, there is no need to pass in the `Subject` it will automatically be filled by the BE.